Riot/Modules/Room/RoomViewController+Spaces.swift

```diff
@@ -0,0 +1,127 @@
+        var eventId: String?
+        var userId: String?
+        var groupId: String?
+        
+        // Check permalink to room or event
+        if pathParams[0] == "room" && pathParams.count >= 2 {
+            
+            // The link is the form of "/room/[roomIdOrAlias]" or "/room/[roomIdOrAlias]/[eventId]"
+            roomIdOrAlias = pathParams[1]
+            
+            // Is it a link to an event of a room?
+            eventId = pathParams.count >= 3 ? pathParams[2] : nil
+            
+        } else if pathParams[0] == "group" && pathParams.count >= 2 {
+            
+            // The link is the form of "/group/[groupId]"
+            groupId = pathParams[1]
+            
+        } else if (pathParams[0].hasPrefix("#") || pathParams[0].hasPrefix("!")) && pathParams.count >= 1 {
+            
+            // The link is the form of "/#/[roomIdOrAlias]" or "/#/[roomIdOrAlias]/[eventId]"
+            // Such links come from matrix.to permalinks
+            roomIdOrAlias = pathParams[0]
+            eventId = pathParams.count >= 2 ? pathParams[1] : nil
+            
+        } else if pathParams[0] == "user" && pathParams.count == 2 { // Check permalink to a user
+            // The link is the form of "/user/userId"
+            userId = pathParams[1]
+        } else if pathParams[0].hasPrefix("@") && pathParams.count == 1 {
+            // The link is the form of "/#/[userId]"
+            // Such links come from matrix.to permalinks
+            userId = pathParams[0]
+        }
+        
+        guard let roomIdOrAlias = roomIdOrAlias else {
+            AppDelegate.theDelegate().handleUniversalLinkURL(url)
+            return
+        }
+        
+        self.startActivityIndicator()
+        
+        var viaServers: [String] = []
+        if let queryParams = queryParams as? [String: Any], let via = queryParams["via"] as? [String] {
+            viaServers = via
+        }
+        
+        if roomIdOrAlias.hasPrefix("#") {
+            self.mainSession.matrixRestClient.roomId(forRoomAlias: roomIdOrAlias) { [weak self] response in
+                guard let self = self else {
+                    return
+                }
+                
+                guard let roomId = response.value else {
+                    self.stopActivityIndicator()
+                    
+                    if response.error != nil {
+                        let errorMessage = VectorL10n.roomDoesNotExist(roomIdOrAlias)
+                        AppDelegate.theDelegate().showAlert(withTitle: nil, message: errorMessage)
+                    }
+                    return
+                }
+                
+                self.requestSummary(forRoomWithId: roomId, via: viaServers, from: url)
+            }
+        } else {
+            self.requestSummary(forRoomWithId: roomIdOrAlias, via: viaServers, from: url)
+        }
+    }
+    
+    private func requestSummary(forRoomWithId roomId: String, via: [String], from url: URL?) {
```

Comment:
We can probably name the method `requestSummaryAndShowSpaceDetail` to be more explicit.